src/python/pants/base/validation.py

```diff
@@ -24,23 +24,36 @@ def assert_list(obj, expected_type=string_types, can_be_none=True, default=(),
   :param raise_type    : the error to throw if the type is not correct.
   """
   val = obj
+  field_msg = ''
   if val is None:
     if can_be_none:
       val = list(default)
     else:
+      if field_name is None:
```

Comment:
You've got this code repeated 3 times in this function. You can set it once at the top (line 27 in this version of the diff)